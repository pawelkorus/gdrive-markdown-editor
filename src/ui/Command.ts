type Command = {
    id: string;
    name: string;
    execute: () => void;
};

export default Command;
