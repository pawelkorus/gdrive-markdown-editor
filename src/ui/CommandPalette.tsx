import React, { useEffect, useState } from 'react';
import { Modal } from 'react-bootstrap';

export type Item = {
    id: string,
    name: string
}

interface CommandPaletteProps {
    show: boolean;
    commands: Item[]
    onItemSelected?: (commandPalleteItem: Item) => void
}

const CommandPalette: React.FC<CommandPaletteProps> = ({show, commands, onItemSelected}) => {
    const [isVisible, setIsVisible] = useState(show);
    const [selected, setSelected] = useState(0);

    useEffect(() => {
        setIsVisible(show);
    }, [show]);

    function onKeydown(e: React.KeyboardEvent<Element>) {
        const key = e.key;
        if (key === "ArrowDown") {
            e.preventDefault();
            setSelected((s) => {
                console.log(s);
                return Math.min(s + 1, commands.length - 1)
            });
            return;
        }
        if (key === "ArrowUp") {
            e.preventDefault();
            setSelected((s) => {
                console.log(s)
                return Math.max(s - 1, 0)
            });
            return;
        }
        if (key === "Enter") {
            e.preventDefault();
            if (onItemSelected) {
                onItemSelected(commands[selected]);
            }
            return;
        }
    }

    return (
        <Modal show={isVisible}>
            <Modal.Body onKeyDown={onKeydown}>
                <div className="list-group">
                {commands.map((item, i) => (
                        <a  key={i.toString()} 
                            href="#" 
                            className={`list-group-item list-group-item-action ${i === selected ? 'active' : ''}`} 
                            aria-current={i === selected ? true : false} 
                            onMouseDown={() => onItemSelected && onItemSelected(item)}
                            onMouseMove={() => setSelected(i)}
                            >
                            {item.name}
                        </a>
                        ))}
                </div>
            </Modal.Body>
        </Modal>
    );
};

export default CommandPalette;
