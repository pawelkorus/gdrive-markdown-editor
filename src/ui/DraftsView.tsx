import React from 'react'
import { useDraftFiles } from '../service/draftfile'
import { useGdriveFile } from '../service/gdrivefile'
import { useNavigateTo } from '../service/navigate'

export default function (): React.ReactElement {
  const [fileDetails] = useGdriveFile()
  const {
    draftFiles,
    discardDraft,
  } = useDraftFiles(fileDetails)
  const { navigateToFileEdit } = useNavigateTo()

  const onUseDraft = (draftId: string) => {
    navigateToFileEdit({ fileId: fileDetails.id, draftId: draftId })
  }

  const onDiscardDraft = (draftId: string) => {
    discardDraft(draftId)
  }

  return (
    <div className="container-lg mt-4">
      <div className="row">
        {draftFiles.length === 0 && (
          <div className="alert alert-info" role="alert">
            No drafts available.
          </div>
        )}
        { draftFiles.length > 0 && (
          <div className="list-group">
            {draftFiles.map(draftFile => (
              <div key={draftFile.id} className="list-group-item list-group-item-action">
                {draftFile.id}
                <button className="btn btn-outline-primary" onClick={() => onUseDraft(draftFile.id)}>Use</button>
                <button className="btn btn-outline-danger" onClick={() => onDiscardDraft(draftFile.id)}>Discard</button>
              </div>
            ))}
          </div>
        )}
      </div>
    </div>
  )
}
