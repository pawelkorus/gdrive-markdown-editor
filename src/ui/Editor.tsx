import React from 'react'
import { useState } from 'react'
import { MarkdownContent } from "../markdown"
import { showPicker } from '../google'

type Props = {
    content: string,
    onSaveClicked?: (event:SaveEvent) => void
    onCloseClicked?: () => void
    onTogglePreviewClicked?: () => void
}

export class SaveEvent {
    constructor(
        public content:string
    ) {}
}

export default function(props:Props):React.ReactElement {
    const [previewEnabled, setPreviewEnabled] = useState(false)
    const [updatedContent, setUpdatedContent] = useState(props.content)

    function togglePreview() {
        setPreviewEnabled(!previewEnabled);
        props.onTogglePreviewClicked && props.onTogglePreviewClicked();
    }

    function browseGdrive() {
        showPicker().then(res => setUpdatedContent(updatedContent + res))
    }

    return (
<div>
    <div className="container-fluid p-2">
        <div className="d-flex flex-row">
            <div className="me-auto me-2"></div>
            <button className="btn btn-primary ms-1" id="btn-save" type="button" onClick={e => props.onSaveClicked(new SaveEvent(updatedContent))}>Save</button>
            <button className="btn btn-primary ms-1" id="btn-browse" type="button" onClick={browseGdrive}>Browse</button>
            <button className="btn btn-primary ms-1" id="btn-preview" type="button" onClick={togglePreview}>Preview</button>
            <button className="btn btn-primary ms-1" id="btn-close" type="button" onClick={props.onCloseClicked}>Close</button>
        </div>
    </div>
    <div className="container-lg mt-4" id="main">
        <div className="row content"> 
            <div className="col">
                    <textarea className="form-control w-100 h-100" 
                        value={updatedContent} 
                        onChange={e => { setUpdatedContent(e.target.value) }}></textarea>
            </div>

            {previewEnabled && <div className="col"><MarkdownContent content={updatedContent}/></div>}    
        </div>
    </div>
</div>
)}
