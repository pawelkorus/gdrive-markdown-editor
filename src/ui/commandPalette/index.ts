export { default as CommandPalette } from './CommandPalette'
export { CommandPaletteProvider } from './CommandPaletteProvider'
export { useCommandPalette } from './useCommandPalette'
