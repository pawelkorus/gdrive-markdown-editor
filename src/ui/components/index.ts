export { default as IntegrationCheck } from './IntegrationCheck'
export { default as DraftSelectorPanel } from './DraftSelectorPanel'
export { default as LastSavedTimestampPanel } from './LastSavedTimestampPanel'
