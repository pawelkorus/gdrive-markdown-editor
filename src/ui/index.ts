export { default as EditorView } from './EditorView'
export { default as ViewerView } from './ViewerView'
export { default as NotificationView } from './NotificationView'
export { default as SourceView } from './SourceView'
export { default as useGlobalCommands } from './useGlobalCommands'
