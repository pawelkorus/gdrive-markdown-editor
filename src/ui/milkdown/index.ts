export { default as MilkdownEditor } from './MilkdownEditor'
export { default as WrapWithProviders } from './WrapWithProviders'
