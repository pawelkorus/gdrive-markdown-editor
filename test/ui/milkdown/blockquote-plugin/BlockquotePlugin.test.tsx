import { MilkdownEditor } from '@app/ui/milkdown'
import { MilkdownProvider } from '@milkdown/react'
import { ProsemirrorAdapterProvider } from '@prosemirror-adapter/react'
import { act, render } from '@testing-library/react'
import React, { PropsWithChildren } from 'react'
import { expect, test } from 'vitest'

test('should render blockquote with additional blockquote class for bootstrap', async () => {
  const SAMPLE_CONTENT = `> blockquote`
  const container = await act(async () => {
    const { container } = render(<WrapWithProviders><MilkdownEditor content={SAMPLE_CONTENT} /></WrapWithProviders>)

    await new Promise<HTMLElement>(r => setTimeout(r, 100))
    return container
  })

  const blockquoteElement = container.querySelector('blockquote')
  expect(blockquoteElement).toHaveClass('blockquote')
  expect(document.body).toMatchSnapshot()
})

const WrapWithProviders: React.FC<PropsWithChildren> = ({ children }) => {
  return (
    <MilkdownProvider>
      <ProsemirrorAdapterProvider>
        {children}
      </ProsemirrorAdapterProvider>
    </MilkdownProvider>
  )
}
