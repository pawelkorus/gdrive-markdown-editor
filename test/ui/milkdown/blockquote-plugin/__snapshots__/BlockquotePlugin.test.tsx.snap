// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render blockquote with additional blockquote class for bootstrap 1`] = `
<body>
  <script
    defer=""
    src="https://accounts.google.com/gsi/client"
  />
  <script
    defer=""
    src="https://apis.google.com/js/api.js"
  />
  <div>
    <div
      data-milkdown-root="true"
    >
      <div
        class="milkdown"
      >
        <div
          class="ProseMirror editor"
          contenteditable="true"
          role="textbox"
          translate="no"
        >
          <blockquote
            class="blockquote"
          >
            <p>
              blockquote
            </p>
          </blockquote>
        </div>
      </div>
    </div>
  </div>
</body>
`;
